torch/testing/_internal/common_utils.py

```diff
@@ -1773,6 +1778,21 @@ def make_tensor(size, device: torch.device, dtype: torch.dtype, *, low=None, hig
         result = torch.repeat_interleave(result, 2, dim=-1)
         result = result[..., ::2]
 
+    if exclude_values:
+        if dtype in integral_types() or dtype is torch.bool:
+            replace_with = torch.tensor(1, device=device, dtype=dtype)
+        elif dtype in floating_types_and(torch.half, torch.bfloat16):
+            replace_with = torch.tensor(torch.finfo(dtype).eps, device=device, dtype=dtype)
+        else:
+            # An assert is required? It's already done above (L1766)
+            # assert dtype in complex_types()
+            float_dtype = torch.float if dtype is torch.cfloat else torch.double
+            real = torch.tensor(torch.finfo(float_dtype).eps, device=device, dtype=dtype)
+            imag = torch.tensor(torch.finfo(float_dtype).eps, device=device, dtype=dtype)
+            replace_with = torch.complex(real, imag)
+        for exclude_val in exclude_values:
+            result[result == exclude_val] = replace_with
```

Comment:
Should we always replace the value with `eps` or maybe do `original_value + eps`?
I favour the second one, this way we can check values close to the different poles.

Eg,
`0` -> `0 + eps`
`-1` -> `-1 + eps`

Reply:
I agree! This is great point, should we change this for integer inputs as well? `original_value + 1`? I think for complex and float tensors, `original_value + torch.finfo(float_dtype).eps` makes sense.

Reply:
I think we should do the same for integer inputs for consistency.

cc: @mruberry what do you think about this?

Reply:
I agree. I've added for integer inputs, just for boolean inputs there needs to be a check. Fixed in recent commit.